File: app/src/main/java/com/example/jpalvarezl/abstractadapter/base/ViewHolderItem.java
package com.example.jpalvarezl.abstractadapter.base;


import android.support.v7.widget.RecyclerView;
import android.view.View;

abstract class ViewHolderItem<T> extends RecyclerView.ViewHolder{
    public ViewHolderItem(View itemView) {
        super(itemView);
    }

    public abstract void bind(T item);
}

File: app/src/main/java/com/example/jpalvarezl/abstractadapter/base/ViewHolderItem.kt
package com.example.jpalvarezl.abstractadapter.base


import android.support.v7.widget.RecyclerView
import android.view.View

abstract class ViewHolderItem<in T>(protected var view: View) : RecyclerView.ViewHolder(view) {
    abstract fun bind(item: T)
}
